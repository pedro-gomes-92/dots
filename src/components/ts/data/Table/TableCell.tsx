import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { Base, BaseProps } from '../../Base';
import { Text } from '../../text';

export interface TableCellProps extends BaseProps {
  text: string;
}

export const TableCell: FunctionComponent<TableCellProps> = ({
  className,
  text,
  ...rest
}: TableCellProps): JSX.Element => {
  return (
    <Base className={classnames('table-cell', className)} {...rest}>
      <Text text={text} />
    </Base>
  );
};

TableCell.defaultProps = {
  tag: 'td',
};
