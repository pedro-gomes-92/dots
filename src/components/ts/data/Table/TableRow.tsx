import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { BaseProps, Base } from '../../Base';
import { Children } from '../../typings';
import { TableCell } from './TableCell';

export interface TableRowProps extends BaseProps {
  children: Children<typeof TableCell>;
}

export const TableRow: FunctionComponent<TableRowProps> = ({
  className,
  children,
  ...rest
}: TableRowProps): JSX.Element => {
  return (
    <Base className={classnames('table-row', className)} {...rest}>
      {children}
    </Base>
  );
};

TableRow.defaultProps = {
  tag: 'tr',
};
