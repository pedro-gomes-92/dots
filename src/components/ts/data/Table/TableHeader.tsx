import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { BaseProps, Base } from '../../Base';
import { Children } from '../../typings';
import { TableRow } from './TableRow';

export interface TableHeaderProps extends BaseProps {
  children: Children<typeof TableRow>;
}

export const TableHeader: FunctionComponent<TableHeaderProps> = ({
  className,
  children,
  ...rest
}: TableHeaderProps): JSX.Element => {
  return (
    <Base className={classnames('table-header', className)} {...rest}>
      {children}
    </Base>
  );
};

TableHeader.defaultProps = {
  tag: 'thead',
};
