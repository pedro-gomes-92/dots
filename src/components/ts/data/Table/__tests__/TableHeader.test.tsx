import React from 'react';
import { render } from '@testing-library/react';

import { TableHeader } from '../TableHeader';
import { TableRow } from '../TableRow';
import { TableCell } from '../TableCell';

describe('Given <TableHeader />', (): void => {
  let container: HTMLElement;
  const addTableHTML = (content: JSX.Element): JSX.Element => <table>{content}</table>;

  describe('When initializes', (): void => {
    it('Then should match the snapshot', (): void => {
      container = render(
        addTableHTML(
          <TableHeader>
            <TableRow>
              <TableCell text="test" />
            </TableRow>
          </TableHeader>,
        ),
      ).container;
      expect(container).toMatchSnapshot();
    });

    describe('And uses children', (): void => {
      beforeEach((): void => {
        container = render(
          addTableHTML(
            <TableHeader>
              <TableRow>
                <TableCell text="test 1" />
                <TableCell text="test 2" />
              </TableRow>
            </TableHeader>,
          ),
        ).container;
      });

      it('Then should match the snapshot', (): void => {
        expect(container).toMatchSnapshot();
      });

      it('And should render rows', (): void => {
        expect(container.querySelectorAll('.table-header .table-cell').length).toEqual(2);
      });
    });
  });
});
