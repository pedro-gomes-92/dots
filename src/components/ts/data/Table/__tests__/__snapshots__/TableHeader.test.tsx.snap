// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <TableHeader /> When initializes And uses children Then should match the snapshot 1`] = `
<div>
  <table>
    <thead
      class="table-header"
    >
      <tr
        class="table-row"
      >
        <td
          class="table-cell"
        >
          <span
            class="text"
          >
            test 1
          </span>
        </td>
        <td
          class="table-cell"
        >
          <span
            class="text"
          >
            test 2
          </span>
        </td>
      </tr>
    </thead>
  </table>
</div>
`;

exports[`Given <TableHeader /> When initializes Then should match the snapshot 1`] = `
<div>
  <table>
    <thead
      class="table-header"
    >
      <tr
        class="table-row"
      >
        <td
          class="table-cell"
        >
          <span
            class="text"
          >
            test
          </span>
        </td>
      </tr>
    </thead>
  </table>
</div>
`;
